Northstar.Custom/mod/scripts/vscripts/sh_server_dialog.gnut

```diff
@@ -0,0 +1,288 @@
+{
+    AddServerToClientStringCommandCallback( "ServerRequestResquest", ServerCallback_DialogueRequestReceived )
+}
+
+//  string header
+//  string message
+//  string image
+//  string rightImage
+//  bool forceChoice
+//  bool noChoice
+//  bool noChoiceWithNavigateBack
+//  bool showSpinner
+//  bool showPCBackButton
+//  float inputDisableTime
+//  bool darkenBackground
+//  bool useFullMessageHeigh
+
+//  array<ServerDialogButtonData> buttonData
+//  the long array is in this order
+
+void function ServerCallback_DialogueRequestReceived( array<string> args )
+{
+    if ( args.len() != 16 )
+        return
+    
+    string header = StringReplace( args[0], "./s.", " ", true, true )
+    string message = StringReplace( args[1], "./s.", " ", true, true )
+    string image = args[2]
+    string rightImage = args[3]
+    bool forceChoice = args[4].tointeger() == 1
+    bool noChoice = args[5].tointeger()  == 1
+    bool noChoiceWithNavigateBack = args[6].tointeger() == 1
+    bool showSpinner = args[7].tointeger() == 1
+    bool showPCBackButton = args[8].tointeger() == 1
+    float inputDisableTime = args[9].tofloat()
+    bool darkenBackground = args[10].tointeger() == 1
+    bool useFullMessageHeight = args[11].tointeger() == 1
+
+    string button0Text = StringReplace( args[12], "./s.", " ", true, true )
+    string button1Text = StringReplace( args[13], "./s.", " ", true, true )
+    string button2Text = StringReplace( args[14], "./s.", " ", true, true )
+    string button3Text = StringReplace( args[15], "./s.", " ", true, true )
+
+    RunUIScript( 
+        "ClientCallback_RequestDialog",
+        header,
+        message,
+        image,
+        rightImage,
+        forceChoice,
+        noChoice,
+        noChoiceWithNavigateBack,
+        showSpinner,
+        showPCBackButton,
+        inputDisableTime,
+        darkenBackground,
+        useFullMessageHeight,
+        button0Text,
+        button1Text,
+        button2Text,
+        button3Text
+    )
+}
+
+#elseif UI
+
+void function ClientCallback_RequestDialog( string header, string message, string image, string rightImage, bool forceChoice, bool noChoice, bool noChoiceWithNavigateBack, bool showSpinner, bool showPCBackButton, float inputDisableTime, bool darkenBackground, bool useFullMessageHeight, string button0Text, string button1Text, string button2Text, string button3Text )
+{   
+    if ( uiGlobal.activeMenu != null )
+        return
+    
+    if ( cooldown >= Time() )
+        return
+    cooldown = Time() + 1
+
+    DialogData dialogData
+
+    dialogData.header = header
+    dialogData.message = message
+    
+    if ( image.find("/") != null )
+        dialogData.image = StringToAsset( image )
+    if ( rightImage.find("/") != null )
+        dialogData.rightImage = StringToAsset( rightImage )
+
+    dialogData.forceChoice = forceChoice
+    dialogData.noChoice = noChoice
+    dialogData.noChoiceWithNavigateBack = noChoiceWithNavigateBack
```

Comment:
user should always have a choice to exit the dialog right?

Reply:
you can always press `ESC`